### PR TITLE
Fixing SetShardTabletControl blacklisted_tables.

### PR DESCRIPTION
**Pull request by alainjobart**
It was wrong and confusing. Hopefully better now.

### CODE REVIEW

File: go/vt/vtctl/vtctl.go [hunk 1]
```diff
@@ -224,8 +224,11 @@ var commands = []commandGroup{
...
 				"Add or remove served type to/from a shard. This is meant as an emergency function. It does not rebuild any serving graph i.e. does not run 'RebuildKeyspaceGraph'."},
 			{"SetShardTabletControl", commandSetShardTabletControl,
-				"[--cells=c1,c2,...] [--blacklisted_tables=t1,t2,...] [--remove] [--disable_query_service] <keyspace/shard> <tablet type>",
-				"Sets the TabletControl record for a shard and type. Only use this for an emergency fix or after a finished vertical split. The *MigrateServedFrom* and *MigrateServedType* commands set this field appropriately already. Always specify the blacklisted_tables flag for vertical splits, but never for horizontal splits."},
+				"[--cells=c1,c2,...] [--tables=t1,t2,...] [--remove] [--disable_query_service] <keyspace/shard> <tablet type>",
+				"Sets the TabletControl record for a shard and type. Only use this for an emergency fix or after a finished vertical split. The *MigrateServedFrom* and *MigrateServedType* commands set this field appropriately already. Always specify the blacklisted_tables flag for vertical splits, but never for horizontal splits.\n" +
```
Comment (alainjobart) [inline go/vt/vtctl/vtctl.go]: Oh the topo field is also named blacklisted_tables, so I will use that consistently then.

**Comment by alainjobart** [comment]
@michael-berlin PTAL you're right it's better that way.

File: go/vt/vtctl/vtctl.go [hunk 2]
```diff
@@ -1230,9 +1233,9 @@ func commandSetShardServedTypes(ctx context.Context, wr *wrangler.Wrangler, subF
 
 func commandSetShardTabletControl(ctx context.Context, wr *wrangler.Wrangler, subFlags *flag.FlagSet, args []string) error {
 	cellsStr := subFlags.String("cells", "", "Specifies a comma-separated list of cells to update")
-	tablesStr := subFlags.String("tables", "", "Specifies a comma-separated list of tables to replicate (used for vertical split). Each is either an exact match, or a regular expression of the form /regexp/")
-	remove := subFlags.Bool("remove", false, "Removes cells for vertical splits. This flag requires the *tables* flag to also be set.")
-	disableQueryService := subFlags.Bool("disable_query_service", false, "Disables query service on the provided nodes")
+	blacklistedTablesStr := subFlags.String("blacklisted_tables", "", "Specifies a comma-separated list of tables to blacklist (used for vertical split). Each is either an exact match, or a regular expression of the form '/regexp/'.")
+	remove := subFlags.Bool("remove", false, "Removes cells for vertical splits.")
+	disableQueryService := subFlags.Bool("disable_query_service", false, "Disables query service on the provided nodes. This flag requires 'tables' and 'remove' to be unset, otherwise it's ignored.")
```
Comment (michael-berlin) [inline go/vt/vtctl/vtctl.go]: s/tables/blacklisted_tables/

File: go/vt/wrangler/shard.go [hunk 3]
```diff
@@ -83,10 +83,10 @@ func (wr *Wrangler) SetShardServedTypes(ctx context.Context, keyspace, shard str
 //   the TabletControl record for the cells
 //
 // This takes the keyspace lock as to not interfere with resharding operations.
-func (wr *Wrangler) SetShardTabletControl(ctx context.Context, keyspace, shard string, tabletType topodatapb.TabletType, cells []string, remove, disableQueryService bool, tables []string) (err error) {
+func (wr *Wrangler) SetShardTabletControl(ctx context.Context, keyspace, shard string, tabletType topodatapb.TabletType, cells []string, remove, disableQueryService bool, blacklistedTables []string) (err error) {
 	// check input
-	if disableQueryService && len(tables) > 0 {
-		return fmt.Errorf("SetShardTabletControl cannot have both DisableQueryService set and tables set")
+	if disableQueryService && len(blacklistedTables) > 0 {
+		return fmt.Errorf("SetShardTabletControl cannot have both DisableQueryService set and BlacklistedTables set")
```
Comment (michael-berlin) [inline go/vt/wrangler/shard.go]: Remove the first "set" here.

**Pull request merged**